src/optimizer/input_column_deriver.cpp

```diff
@@ -296,6 +298,11 @@ void InputColumnDeriver::JoinHelper(const BaseOperatorNodeContents *op) {
   } else if (op->GetType() == OpType::INNERNLJOIN) {
     auto join_op = reinterpret_cast<const InnerNLJoin *>(op);
     join_conds = join_op->GetJoinPredicates();
+  } else if (op->GetType() == OpType::LEFTSEMIHASHJOIN) {
+    auto join_op = reinterpret_cast<const LeftSemiHashJoin *>(op);
```

Comment:
I think this will call the copu constructor. You should pass these in the constructor itself and initialize in the initializor list.